hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -334,6 +334,19 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		return ctrl.Result{}, fmt.Errorf("failed to look up release image metadata: %w", err)
 	}
 
+	// Validate the HostedCluster payload can support the NodePool's CPU arch
+	if err = validateHCPayloadSupportsNodePoolCPUArch(hcluster, nodePool); err != nil {
+		SetStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+			Type:               hyperv1.NodePoolValidArchPlatform,
+			Status:             corev1.ConditionFalse,
+			Reason:             hyperv1.NodePoolInvalidArchPlatform,
+			Message:            fmt.Sprintf("NodePool CPU arch %s cannot be supported by the HostedCluster payload type: %s", nodePool.Spec.Arch, hcluster.Status.Payload),
```

Comment:
you could just use `err.Error()` as the message here.
How about including the solution in the message? like `either change your NodePool arch or use a multi release image`

Reply:
Updated